tests/test_estimation.py

```diff
@@ -1,91 +1,206 @@
 import numpy as np
-from pyriemann.estimation import (Covariances, ERPCovariances,
-                                  XdawnCovariances, CospCovariances,
-                                  HankelCovariances, Coherences, Shrinkage)
+from pyriemann.estimation import (
+    Covariances,
+    ERPCovariances,
+    XdawnCovariances,
+    CospCovariances,
+    HankelCovariances,
+    Coherences,
+    Shrinkage,
+)
 import pytest
 
 
-def test_covariances():
+estim = ["cov", "scm", "lwf", "oas", "mcd", "corr"]
+coh = ["ordinary", "instantaneous", "lagged", "imaginary"]
+
+
+@pytest.mark.parametrize("estimator", estim)
+def test_covariances(estimator, rndstate):
     """Test Covariances"""
-    x = np.random.randn(2, 3, 100)
-    cov = Covariances()
+    n_trials, n_channels, n_times = 2, 3, 100
+    x = rndstate.randn(n_trials, n_channels, n_times)
+    cov = Covariances(estimator=estimator)
     cov.fit(x)
-    cov.fit_transform(x)
-    assert cov.get_params() == dict(estimator='scm')
+    covmats = cov.fit_transform(x)
+    assert cov.get_params() == dict(estimator=estimator)
+    assert covmats.shape == (n_trials, n_channels, n_channels)
+    assert np.any(np.linalg.eigvals(covmats) > 0.0)
```

Comment:
Why is it `np.any(`), and not `np.all()`?

Reply:
Same question for tests below.

Reply:
Moreover, covariance matrices are SPSD instead of SPD.
Test should be `eigvals >= 0`.